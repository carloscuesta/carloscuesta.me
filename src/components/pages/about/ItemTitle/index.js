// @flow
import React, { type Node, type Element } from 'react'

import styles from './styles.module.css'

type Props = {
  children: string | Node
}

const Title = (props: Props): Element<'p'> => (
  <p className={styles.title}>
    {props.children}
  </p>
)

export default Title
