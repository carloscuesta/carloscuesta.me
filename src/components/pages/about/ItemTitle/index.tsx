import { type ReactNode } from 'react'

import styles from './styles.module.css'

type Props = {
  children: ReactNode
}

const Title = (props: Props) => (
  <p className={styles.title}>
    {props.children}
  </p>
)

export default Title
