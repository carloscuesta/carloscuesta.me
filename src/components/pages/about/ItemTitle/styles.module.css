.title {
  margin: 0;
  font-size: 1.15rem;
  font-weight: 700;
}
