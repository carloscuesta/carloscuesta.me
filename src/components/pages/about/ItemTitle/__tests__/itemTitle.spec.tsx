import renderer from 'react-test-renderer'

import ItemTitle from '../index'
import * as stubs from './stubs'

describe('ItemTitle', () => {
  describe('when children is a string', () => {
    it('should match ItemTitle component', () => {
      const wrapper = renderer.create(<ItemTitle {...stubs.propsString} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when children is a Component', () => {
    it('should match ItemTitle component', () => {
      const wrapper = renderer.create(<ItemTitle {...stubs.propsComponent} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
