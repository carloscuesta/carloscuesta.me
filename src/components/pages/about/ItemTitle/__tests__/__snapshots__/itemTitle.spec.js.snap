// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTitle when children is a Component should match ItemTitle component 1`] = `
<h5
  className="title"
>
  <h1>
    Component
  </h1>
</h5>
`;

exports[`ItemTitle when children is a string should match ItemTitle component 1`] = `
<h5
  className="title"
>
  Title
</h5>
`;
